homeassistant/components/eight_sleep/__init__.py

```diff
@@ -168,9 +180,26 @@ def __init__(
 
         mapped_name = NAME_MAP.get(sensor, sensor.replace("_", " ").title())
         if self._user_obj is not None:
-            mapped_name = f"{self._user_obj.side.title()} {mapped_name}"
-
-        self._attr_name = f"Eight {mapped_name}"
+            self._attr_name = (
+                f"{self._user_obj.user_profile['firstName']}'s {mapped_name}"
+            )
+        else:
+            self._attr_name = f"Eight Sleep {mapped_name}"
         self._attr_unique_id = (
             f"{_get_device_unique_id(eight, self._user_obj)}.{sensor}"
         )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, _get_device_unique_id(eight, self._user_obj))}
+        )
+
+    async def async_heat_set(self, target: int, duration: int) -> None:
+        """Handle eight sleep service calls."""
+        if self._user_obj is not None:
+            await self._user_obj.set_heating_level(target, duration)
+            await self.hass.data[DOMAIN][self._config_entry.entry_id][
+                DATA_HEAT
+            ].async_request_refresh()
+        else:
+            raise HomeAssistantError(
+                "This entity does not support the heat set service."
+            )
```

Comment:
If you reverse the condition and swap these you can raise and then outdent the other block